pkg/workflows/sdk/v2/consensus_aggregators.go

```diff
@@ -0,0 +1,164 @@
+package sdk
+
+import (
+	"fmt"
+	"math/big"
+	"reflect"
+
+	"github.com/shopspring/decimal"
+	"github.com/smartcontractkit/chainlink-common/pkg/workflows/sdk/v2/pb"
+)
+
+type NumericType interface {
+	int | int8 | int16 | int32 | int64 | uint | uint8 | uint16 | uint32 | uint64 | float32 | float64 | *big.Int | decimal.Decimal
+}
+
+type Primitive interface {
+	NumericType | string | bool
+}
+
+func ConsensusMedianAggregation[T NumericType]() ConsensusAggregation[T] {
+	return &consensusDescriptor[T]{Descriptor_: &pb.ConsensusDescriptor_Aggregation{Aggregation: pb.AggregationType_MEDIAN}}
+}
+
+func ConsensusIdenticalAggregation[T any]() ConsensusAggregation[T] {
+	var t T
+	if isIdenticalType(reflect.TypeOf(t)) {
+		return &consensusDescriptor[T]{Descriptor_: &pb.ConsensusDescriptor_Aggregation{Aggregation: pb.AggregationType_IDENTICAL}}
+	}
+
+	return &consensusDescriptorError[T]{Error: fmt.Errorf("%T is not a valid type for identical consensus", t)}
+}
+
+func ConsensusCommonPrefixAggregation[T any]() func() (ConsensusAggregation[[]T], error) {
+	return func() (ConsensusAggregation[[]T], error) {
+		var t []T
+		if isIdenticalSliceOrArray(reflect.TypeOf(t)) {
+			return &consensusDescriptor[[]T]{Descriptor_: &pb.ConsensusDescriptor_Aggregation{Aggregation: pb.AggregationType_COMMON_PREFIX}}, nil
+		}
+
+		return &consensusDescriptor[[]T]{}, fmt.Errorf("%T is not a valid type for common prefix consensus", t)
+	}
+}
+
+func ConsensusCommonSuffixAggregation[T any]() func() (ConsensusAggregation[[]T], error) {
+	return func() (ConsensusAggregation[[]T], error) {
+		var t []T
+		if isIdenticalSliceOrArray(reflect.TypeOf(t)) {
+			return &consensusDescriptor[[]T]{Descriptor_: &pb.ConsensusDescriptor_Aggregation{Aggregation: pb.AggregationType_COMMON_SUFFIX}}, nil
+		}
+
+		return &consensusDescriptor[[]T]{}, fmt.Errorf("%T is not a valid type for common prefix consensus", t)
+	}
+}
+
+func ConsensusAggregationFromTags[T any]() ConsensusAggregation[T] {
+	var zero T
+	t := reflect.TypeOf(zero)
+	descriptor, err := parseConsensusTag(t)
+	if err != nil {
+		return &consensusDescriptorError[T]{Error: err}
+	}
+	return (*consensusDescriptor[T])(descriptor)
+}
+
+var bigIntType = reflect.TypeOf((*big.Int)(nil))
+
+func parseConsensusTag(t reflect.Type) (*pb.ConsensusDescriptor, error) {
+	if t.Kind() == reflect.Pointer {
+		t = t.Elem()
+	}
+
+	if t.Kind() != reflect.Struct {
+		return nil, fmt.Errorf("ConsensusAggregationFromTags expects a struct type, got %s", t.Kind())
+	}
+
+	descriptors := make(map[string]*pb.ConsensusDescriptor)
+	for i := 0; i < t.NumField(); i++ {
+		field := t.Field(i)
+		tag := field.Tag.Get("consensus")
+		if tag == "ignore" || tag == "" {
+			continue
+		}
+
+		tpe := field.Type
+		if tpe.Kind() == reflect.Pointer && tpe != bigIntType {
+			tpe = tpe.Elem()
+		}
+
+		var err error
+		switch tag {
+		case "median":
```

Comment:
nit: those strings should be enums

Reply:
Tags are always strings, and the enums generated by the protos are are ints. Leaving for now.